### ISSUE TITLE
算法细节问题

### ISSUE THREAD

**Issue by WeiCL7777** [wangf3014/SCLIP#3]
作者您好！我对文章中的思路有些疑惑，希望咨询作者。
（1）文中提到将CLIP最后一层修改后作为decoder，进行分割。请问进行zero-shot分割时，还需要text encoder信息吗？还是说经过decoder处理后直接输出分割结果了？因为文中没有给出分割的整体框图，所以我对这个分割流程不太清晰。
（2）另外，请问作者可以将修改后的CLIP作为一个特征提取backbone，后面加上一些分割head，在下游任务fine tune后得到一个整体模型吗？这种思路是否可行？
谢谢作者，期待您的回复！

**Comment by wangf3014** [comment]
Hi, thank you for reaching out.

SCLIP needs text inputs as well as a text encoder to perform segmentation (patch-wise classification). Specifically, given an input image in 3*224*224 size, the vision encoder produces d*14*14 features (Fi); and given a k-class segmentation task, we have k textual class names (supposing one name for each class) and can obtain a k*d matrix (Ft) by the text encoder. We then get the patch-wise logits (k*14*14) by projecting Fi with Ft.

Fine-tuning a task-specific decoder head on top of the modified CLIP backbone can definately achieve much higher results than the zero-shot model. However, in this way the model will lose its capabilities in open-vocabulary inference.
